controllers/upgrade_handlers.go

```diff
@@ -18,39 +18,125 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// handleUpgrade orchestrate main upgrade steps
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	// log and update status to in-progress
+	r.Log.Info("Starting handleUpgrade")
+
+	// Pre-pivot steps
+
+	// backup with OADP
+	r.Log.Info("Starting backup")
+	backupRequeue, backupStatus, err := r.BackupRestore.ReconcileBackup(ctx, ibu.Spec.OADPContent)
+	if err != nil {
+		r.Log.Error(err, fmt.Sprintf("error msg from backup: %s, will requeue after: %s", backupStatus.Message, backupRequeue.RequeueAfter))
+		return backupRequeue, err
+	}
+	switch backupStatus.Status {
+	case backuprestore.BackupCompleted:
+		// only proceed when BackupCompleted
+		r.Log.Info(fmt.Sprintf("Backup completed with msg %s", backupStatus.Message))
+	default:
+		// backup in progress
```

Comment:
Don't forget to include the cases BackupFailed(failed upgrade, no requeue), BackupFailedValidation(requeue)